velox/exec/tests/MultiFragmentTest.cpp

```diff
@@ -787,3 +787,133 @@ TEST_F(MultiFragmentTest, earlyCompletionMerge) {
     ASSERT_TRUE(waitForTaskCompletion(task.get())) << task->taskId();
   }
 }
+
+class CustomSlowNode : public core::PlanNode {
+ public:
+  CustomSlowNode(const core::PlanNodeId& id, core::PlanNodePtr source)
+      : PlanNode(id), sources_{std::move(source)} {}
+
+  const RowTypePtr& outputType() const override {
+    return sources_[0]->outputType();
+  }
+
+  const std::vector<core::PlanNodePtr>& sources() const override {
+    return sources_;
+  }
+
+  std::string_view name() const override {
+    return "custom slow";
+  }
+
+ private:
+  void addDetails(std::stringstream& /* stream */) const override {}
+
+  std::vector<core::PlanNodePtr> sources_;
+};
+
+class CustomSlowOperator : public Operator {
+ public:
+  CustomSlowOperator(
+      int32_t operatorId,
+      DriverCtx* driverCtx,
+      std::shared_ptr<const CustomSlowNode> slowNode)
+      : Operator(
+            driverCtx,
+            slowNode->outputType(),
+            operatorId,
+            slowNode->id(),
+            "CustomSlowOperator") {}
+
+  void addInput(RowVectorPtr input) override {
+    inputs_.emplace_back(std::move(input));
```

Comment:
needsInput() returns false if there is at least one entry in 'inputs_', Hence, 'inputs_' will never have more than one entry. Hence, we don't need an std::vector for 'inputs_' and can store the input in 'input_' variable in the base class.